cirq/google/op_serializer.py

```diff
@@ -0,0 +1,164 @@
+# Copyright 2019 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Callable, cast, Dict, List, NamedTuple, Type, TypeVar, Union
+
+import sympy
+
+from cirq import devices, ops
+from cirq.google import arg_func_langs
+
+# Type for variables that are subclasses of ops.Gate.
+Gate = TypeVar('Gate', bound=ops.Gate)
+
+
+class SerializingArg(
+        NamedTuple('SerializingArg',
+                   [('serialized_name', str),
+                    ('serialized_type', Type[arg_func_langs.ArgValue]),
+                    ('gate_getter',
+                     Union[str, Callable[[ops.Gate], arg_func_langs.ArgValue]]),
```

Comment:
We can easily update this to deal with ops when we need to.   However I suspect that the hardware side will not be fundamentally tied to "operations that depend on qubits" any time soon, because that's not how you build a modular system.